gulpfile.js

```diff
@@ -0,0 +1,381 @@
+      .pipe(gulpSourcemaps.write(dest))
+      .pipe(gulp.dest(dest));
+  };
+}
+
+
+/**
+ * Create a Gulp task that executes a process.
+ */
+function createExecTask(packageName, executable, args) {
+  if (!args) {
+    args = executable;
+    executable = undefined;
+  }
+  return function(done) {
+    resolveBin(packageName, { executable: executable }, function (err, binPath) {
+      if (err) {
+        console.error(err);
+        throw err;
+      }
+
+      const spawn = child_process.spawn(binPath, args);
+
+      spawn.stdout.on('data', (data) => {
+        process.stdout.write(data);
+      });
+
+      spawn.stderr.on('data', (data) => {
+        process.stderr.write(data);
+      });
+
+      spawn.on('close', (code) => {
+        if (code != 0) {
+          throw new Error('Process failed with code ' + code);
+        }
+        done();
+      });
+    });
+  }
+}
+
+/***************************************************************************************************
+ * Components Build Tasks.
+ */
+gulp.task(':build:components:ts', createTsBuildTask({ tsConfigPath: componentsDir }));
+gulp.task(':build:components:assets', function() {
+  return gulp.src(path.join(componentsDir, '*/**/*.!(ts|spec.ts)'))
+    .pipe(gulp.dest(outLibDir));
+});
+gulp.task(':build:components:scss', function() {
+  const cssTask = createSassBuildTask({
+    dest: outLibDir,
+    root: componentsDir,
+    includePaths: path.join(componentsDir, 'core/style')
+  });
+  // Also copy over the SCSS for the components.
+  return gulpMerge([
+    cssTask(),
+    gulp.src(path.join(componentsDir, '**/*.scss'))
+      .pipe(gulp.dest(outLibDir))
+  ]);
+});
+gulp.task('build:components', [
+  ':build:components:ts',
+  ':build:components:assets',
+  ':build:components:scss'
+], function() {
+  inlineResources([outLibDir]);
+});
+gulp.task(':build:components:ngc', ['build:components'], createExecTask(
+  '@angular/compiler-cli', 'ngc', ['-p', path.relative(__dirname, componentsDir)]
+));
+
+
+/***************************************************************************************************
+ * Vendor files Build Tasks.
+ */
+gulp.task(':build:vendor', function() {
+  const npmVendorFiles = [
+    '@angular', 'core-js/client', 'hammerjs', 'rxjs', 'systemjs/dist', 'zone.js/dist'
+  ];
+
+  return gulpMerge(
+    npmVendorFiles.map(function(root) {
+      const glob = path.join(root, '**/*.+(js|js.map)');
+      return gulp.src(path.join('node_modules', glob))
+        .pipe(gulp.dest(path.join('dist/vendor', root)));
+    }));
+});
+
+
+/***************************************************************************************************
+ * DevApp Build Tasks.
+ */
+gulp.task(':build:devapp:ts', [':build:components:ts'], createTsBuildTask({ tsConfigPath: devAppDir }));
+gulp.task(':build:devapp:scss', [':build:components:scss'], createSassBuildTask({
+  dest: outDir,
+  root: devAppDir,
+  // Change this once we have a better strategy for releasing SCSS files.
```

Comment:
I think we should omit this for now. Currently the demo-app doesn't use any functions, mixins, etc. from the library.